tool/perftest_providers.py

```diff
@@ -0,0 +1,199 @@
+        'mistralai': ProviderConfig(
+            prompt_udf=create_chatgpt_prompt,
+            udf=pxtf.mistralai.chat_completions,
+            default_model='mistral-tiny',
+            kwargs={'model_kwargs': {'max_tokens': 100, 'temperature': 0.7}},
+        ),
+        'together': ProviderConfig(
+            prompt_udf=create_simple_prompt,
+            udf=pxtf.together.completions,
+            default_model='mistralai/Mixtral-8x7B-Instruct-v0.1',
+            kwargs={'model_kwargs': {'max_tokens': 100, 'temperature': 0.7}},
+        ),
+        'deepseek': ProviderConfig(
+            prompt_udf=create_chatgpt_prompt,
+            udf=pxtf.deepseek.chat_completions,
+            default_model='deepseek-chat',
+            kwargs={'model_kwargs': {'max_tokens': 100, 'temperature': 0.7}},
+        ),
+        'bedrock': ProviderConfig(
+            prompt_udf=create_simple_messages_prompt,
+            udf=pxtf.bedrock.converse,
+            default_model='anthropic.claude-3-haiku-20240307-v1:0',
+            kwargs={'model_kwargs': {'max_tokens': 100, 'temperature': 0.7}},
+        ),
+    }
+
+
+def main() -> None:
+    """Main function to run the test."""
+    parser = argparse.ArgumentParser(
+        description='Test endpoint providers for sentence generation in Pixeltable',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        epilog="""
+Examples:
+  %(prog)s --provider openai --n 10
+  %(prog)s --provider gemini --n 5 --model gemini-1.5-pro
+  %(prog)s --provider anthropic --n 20
+        """,
+    )
+
+    provider_configs = create_provider_configs()
+
+    parser.add_argument(
+        '--provider',
+        required=True,
+        choices=list(provider_configs.keys()),
+        help='AI provider to use for sentence generation',
+    )
+    parser.add_argument('--n', type=int, required=True, help='Number of word pairs to generate sentences for')
+    parser.add_argument('--model', help='Model to use (overrides provider default)')
+    parser.add_argument('--log-level', type=int, default=10, help='Logging level (default: 10)')
+    args = parser.parse_args()
+
+    # Load wordlist
+    with open('/usr/share/dict/american-english', encoding='utf-8') as f:
+        wordlist = [word.strip() for word in f]
+
+    pxt.configure_logging(level=args.log_level)
+    provider_config = provider_configs[args.provider]
+    model = args.model or provider_config.default_model
+
+    print(f'Using provider: {args.provider}')
+    print(f'Using model: {model}')
+    print(f'Generating {args.n} sentences...')
+
+    t = pxt.create_table('sentence_tbl', {'word1': pxt.String, 'word2': pxt.String}, if_exists='replace')
+    t.add_computed_column(prompt=provider_config.prompt_udf(t.word1, t.word2))
+    t.add_computed_column(response=provider_config.udf(t.prompt, model=model, **provider_config.kwargs))
+
+    rows = ({'word1': words[0], 'word2': words[1]} for words in (random.sample(wordlist, k=2) for _ in range(args.n)))
```

Comment:
Just a cute little syntactic convenience you can use here,
```
rows = ({'word1': word1, 'word2': word2} for word1, word2 in (random.sample(wordlist, k=2) for _ in range(args.n)))
```
you avoid the awkward use of `0` and `1` indices which don't really mean anything in this (tuple) context